test/SlowTests/Issues/RavenDB_19548.cs

```diff
@@ -0,0 +1,176 @@
+                 */
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<User>()
+                        let ret = RavenQuery.Raw<int>("result.Id.length")
+                        select ret;
+
+                    Action act = () => asyncDocumentQuery.ToString();
+                    var exception = Assert.Throws<NotSupportedException>(act);
+                    Assert.Equal("Unsupported parameter type Int32. Primitive types/string types are not allowed in Raw<T>() method without wrapping them as an object.",
+                        exception.InnerException.Message);
+                }
+
+                /* should work:
+                 * object type X 'select new'
+                 */
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<User>()
+                        let ret = RavenQuery.Raw<object>("result.Id.length")
+                        select new { myval = ret };
+
+                    Assert.Equal(
+                        "declare function output(result) {\r\n\tvar ret = result.Id.length;\r\n\treturn { myval : ret };\r\n}\r\nfrom 'Users' as result select output(result)"
+                        , asyncDocumentQuery.ToString());
+                }
+
+                /* should work:
+                 * object type X 'select' with JS object inside Raw() function
+                 */
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<User>()
+                        let ret = RavenQuery.Raw<object>("{ retval : result.Id.length}")
+                        select ret;
+
+                    Assert.Equal(
+                        "declare function output(result) {\r\n\tvar ret = { retval : result.Id.length};\r\n\treturn ret;\r\n}\r\nfrom 'Users' as result select output(result)"
+                        , asyncDocumentQuery.ToString());
+                }
+
+                /* shouldn't work:
+                 * object type X 'select'  
+                 * server throws InvalidOperationException: "Query returning a single function call result must return an object"
+                 */
+                using (var session = store.OpenAsyncSession())
+                {
+                    var asyncDocumentQuery = from result in session.Query<User>()
+                        let ret = RavenQuery.Raw<object>("result.Id.length")
+                        select ret;
+
+                    Assert.Equal(
+                        "declare function output(result) {\r\n\tvar ret = result.Id.length;\r\n\treturn ret;\r\n}\r\nfrom 'Users' as result select output(result)"
+                        , asyncDocumentQuery.ToString());
+
+                }
```

Comment:
assert that query execution throws the expected error

Reply:
Done